spec/unit/settings_spec.rb

```diff
@@ -221,6 +221,27 @@ def metadata(setting)
       @settings.define_settings :main, :bool => { :type => :boolean, :default => true, :desc => "desc" }
     end
 
+    it "the cache should be threadsafe" do
+      original_cache = @settings.instance_variable_get(:@cache).value
+      other_request = Thread.new do
+        expect(@settings.instance_variable_get(:@cache).value).not_to be(original_cache)
+      end
+      other_request.join
+      expect(@settings.instance_variable_get(:@cache).value).to be(original_cache)
+    end
+
+    it "the cache can be updated with content to be shared with new threads" do
+      @settings[:myval]
```

Comment:
What is this doing?

Reply:
I've added a comment to the test about this to make it clearer, but to address your question: reading settings adds that setting to the cache, so the test does that to populate the cache and then makes a new thread and ensures that thread gets the new value from the parent and can operate on the cache independently from the parent thread.

Reply:
gotcha, thanks!